### PR TITLE
ruff, ruff-{check,format}: add page

### LINKED ISSUE #12365: Page request: ruff
### Command description

An extremely fast Python linter and code formatter, written in Rust.

### Command details

_No response_

### Documentation

https://docs.astral.sh/ruff/

### Platform

Common

### VCS repository link (e.g. GitHub, GitLab)

https://github.com/astral-sh/ruff

### Additional information

24k Github stars

### PR DESCRIPTION
**Pull request by acuteenvy**
- [x] The page(s) are in the correct platform directories: `common`, `linux`, `osx`, `windows`, `sunos`, `android`, etc.
- [x] The page(s) have at most 8 examples.
- [x] The page description(s) have links to documentation or a homepage.
- [x] The page(s) follow the [content guidelines](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#guidelines).
- [x] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#commit-message-and-pr-title).

Closes #12365.

### CODE REVIEW

**Review by kbdharun — APPROVED** [review]
LGTM, Thanks for your contribution.

File: pages/common/ruff-format.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+- Format given files or directories in-place:
+
+`ruff format {{path/to/file_or_directory1 path/to/file_or_directory2 ...}}`
+
+- Print which files would have been modified and return a non-zero exit code if there are files to reformat, and zero otherwise:
+
+`ruff format --check`
+
+- Print what changes would be made, without modifying the files:
```
Comment (gutjuri) [inline pages/common/ruff-format.md:15]:
```suggestion
- Print what changes would be made without modifying the files:
```

File: pages/common/ruff.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# ruff
+
+> An extremely fast Python linter and code formatter, written in Rust.
```
Comment (gutjuri) [inline pages/common/ruff.md:3]:
```suggestion
> An extremely fast Python linter and code formatter written in Rust.
```

File: pages/common/ruff-check.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+> An extremely fast Python linter. `check` is the default command - it can be omitted everywhere.
+> If no files or directories are specified, the current working directory is used by default.
+> More information: <https://docs.astral.sh/ruff/linter>.
+
+- Run the linter on the given files or directories:
+
+`ruff check {{path/to/file_or_directory1 path/to/file_or_directory2 ...}}`
+
+- Apply suggested fixes:
```
Comment (gutjuri) [inline pages/common/ruff-check.md:11]:
```suggestion
- Apply the suggested fixes in-place:
```

Am I correct in assuming that this modifies the file in-place?

**Review by gutjuri — COMMENTED** [review]
Thank you for your contribution. I have some comments.

File: pages/common/ruff.md (hunk 2 above)
Comment (acuteenvy) [reply pages/common/ruff.md:3]: I copied this from [the official GitHub repo](https://github.com/astral-sh/ruff).

**Review by gutjuri — APPROVED** [review]

**Pull request merged**